.github/workflows/static_analysis.yml

```diff
@@ -0,0 +1,33 @@
+name: Static analysis
+on:
+  pull_request:
+
+jobs:
+  static_analysis:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+
+    - uses: actions/setup-java@v1
+      with:
+        java-version: '12.x'
+
+    - name: Set up Dart SDK
+      uses: dart-lang/setup-dart@v1
+      with:
+        dart-sdk: '2.17.6'
+
+    - uses: subosito/flutter-action@v1
+      with:
+        channel: 'stable'
+        flutter-version: '3.0.0'   
+
+    - name: Get flutter dependencies
+      run: flutter pub get
+
+    - name: Statically analyze the Dart code for any errors/warnings, ignoring "info" level warnings
+      run: |
+        if flutter analyze 2>&1 | grep -q -E 'error:|warning:'; 
```

Comment:
Could you try adding:  `--no-pub` (`flutter analyze --no-pub`) to skip running `pub get` again?